breaking change on unexported field in pull request #36

hi,

i had this struct working before pr#36:
```go
type Config struct {
	MongodbURL   string `env:"MONGODB_URL" envDefault:"localhost:27017"`
	MongodbDb    string `env:"MONGODB_DB" envDefault:"ms-td-dispatch"`
	MqttBroker   string `env:"MQTT_BROKER" envDefault:"tcp://192.168.101.2:9601"`
	HTTPAddr     string `env:"HTTP_ADDR" envDefault:""`
	HTTPPort     string `env:"HTTP_PORT" envDefault:"9500"`
	LogLevel     string `env:"LOG_LEVEL" envDefault:""`
	LogFormatter string `env:"LOG_FORMATTER" envDefault:""`
	logger       *logrus.Logger
}

cfg := &Config{}
err := env.Parse(cfg)
```
but now it is failing with: 
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 1 [running]:
reflect.valueInterface(0x820920, 0xc42007c670, 0x1b6, 0x1, 0x1, 0x0)
        /usr/lib/go-1.8/src/reflect/value.go:936 +0x1a5
reflect.Value.Interface(0x820920, 0xc42007c670, 0x1b6, 0x7, 0x820920)
        /usr/lib/go-1.8/src/reflect/value.go:925 +0x44
github.com/caarlos0/env.doParse(0x802160, 0xc42007c600, 0x199, 0x802160, 0xc42007c600)
        /home/juan/importante/proyectos/monorepo/go/src/github.com/caarlos0/env/env.go:49 +0x5ed
github.com/caarlos0/env.Parse(0x7b6000, 0xc42007c600, 0x20, 0xc4200cd160)
        /home/juan/importante/proyectos/monorepo/go/src/github.com/caarlos0/env/env.go:40 +0xbd

--- a/env.go
+++ b/env.go
@@ -45,7 +45,7 @@ func doParse(ref reflect.Value) error {
 	var errorList []string
 
 	for i := 0; i < refType.NumField(); i++ {
-		if reflect.Ptr == ref.Field(i).Kind() {
+		if reflect.Ptr == ref.Field(i).Kind() && ref.Field(i).CanSet() {
 			err := Parse(ref.Field(i).Interface())
 			if nil != err {
 				return err

--- a/env_test.go
+++ b/env_test.go
@@ -31,6 +31,7 @@ type Config struct {
 
 type ParentStruct struct {
 	InnerStruct *InnerStruct
+	unexported  *InnerStruct
 }
 
 type InnerStruct struct {
@@ -77,16 +78,14 @@ func TestParsesEnv(t *testing.T) {
 }
 
 func TestParsesEnvInner(t *testing.T) {
-
 	os.Setenv("innervar", "someinnervalue")
 	defer os.Clearenv()
-	cfg := ParentStruct{&InnerStruct{}}
+	cfg := ParentStruct{&InnerStruct{}, &InnerStruct{}}
 	assert.NoError(t, env.Parse(&cfg))
 	assert.Equal(t, "someinnervalue", cfg.InnerStruct.Inner)
 }
 
 func TestParsesEnvInnerNil(t *testing.T) {
-
 	os.Setenv("innervar", "someinnervalue")
 	defer os.Clearenv()
 	cfg := ParentStruct{}

